src/deephyper/hpo/_search.py

```diff
@@ -504,6 +522,10 @@ def search(
         else:
             df_results = self.history.to_dataframe()
 
+        # Log calls using json.dumps for formatting
```

Comment:
Please move this before `self._search` is being called. The idea here is to log the input of the `search.search(max_evals, timeout, max_evals_strict)` before running it. However, maybe after the `self._search` is done we can log what actually happened:

```python
logging.info(f"Search will run for {x} eval(s) and {t} time.")
...
self._search(...)
...
logging.info(f"The search has completed after {x} eval(s) and {t} time.")
```